cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -5,6 +5,10 @@ private import DataFlowDispatch
 /** Gets the instance argument of a non-static call. */
 private Node getInstanceArgument(Call call) {
   result.asExpr() = call.getQualifier()
+  or
+  // For constructors, there is no qualifier, so we pretend the call itself
+  // is the instance argument.
+  result.asExpr() = call.(ConstructorCall)
```

Comment:
This should rather be an implicit pre-update node associated with the constructor call.